Makefile

```diff
@@ -1,11 +1,16 @@
 .PHONY: install test clean
 
-install-dev:
-	pip install -e .
-	pip install -r requirements-test.txt
+requirements-test.txt:
+	uv pip compile --group dev pyproject.toml -o requirements-test.txt
+
+requirements.txt:
+	uv pip compile pyproject.toml -o requirements.txt
+
+install: requirements-test.txt
+	uv sync
```

Comment:
I think I'm just unfamiliar with that's happening here. Can you explain why we're using `requirements-test.txt` and `requirements.txt` that are auto-generated from `uv`? Is it just to maintain backward compat for ppl who don't want to use `uv`? I previously assume `uv.lock` kinda replaced these methods.

Maybe you could explain to me, and I can use your explanation to write inline comments to the makefile that would have helped me understand :)

Reply:
I'm all for keeping them for backward compat, especially if core maintainers aren't quite sold on `uv` yet :)

> * (best reason) currently, `dependabot` can't scan `uv.lock` or `pyproject.toml` - so having these files present allows for security scans to occur which effectively check on what's in `uv.lock`

but the "dependabot" rationale feels less strong. it is a bit neglected by github imho and highly locked to github infra, whereas renovate is open source and multi-platform (and is officially what uv recommends)

https://docs.astral.sh/uv/guides/integration/dependency-bots/
https://docs.renovatebot.com/bot-comparison/

Reply:
that's a good point.